glib/src/thread_guard.rs

```diff
@@ -111,4 +115,39 @@ impl<T> Drop for ThreadGuard<T> {
     }
 }
 
+impl<T: Default> Default for ThreadGuard<T> {
+    fn default() -> Self {
+        Self::new(T::default())
+    }
+}
+
 unsafe impl<T> Send for ThreadGuard<T> {}
+
+impl<T: ToValue + StaticType> ToValue for ThreadGuard<T> {
+    fn to_value(&self) -> Value {
+        T::to_value(self.get_ref())
+    }
+
+    fn value_type(&self) -> Type {
+        T::static_type()
+    }
+}
+
+impl<T: From<Value> + ToValue> From<ThreadGuard<T>> for Value {
+    fn from(guard: ThreadGuard<T>) -> Self {
+        Self::from(guard.get_ref())
```

Comment:
```suggestion
        Self::from(guard.into_inner())
```

Reply:
The compiler says that it wants something like this, but its too late at night for me to understand how we are going from `T: From<Value> where Value: From<T>`

```
diff --git a/glib/src/thread_guard.rs b/glib/src/thread_guard.rs
index 1dac4d84686..333789cd724 100644
--- a/glib/src/thread_guard.rs
+++ b/glib/src/thread_guard.rs
@@ -133,9 +133,12 @@ impl<T: ToValue + StaticType> ToValue for ThreadGuard<T> {
     }
 }
 
-impl<T: From<Value> + ToValue> From<ThreadGuard<T>> for Value {
+impl<T: From<Value>> From<ThreadGuard<T>> for Value
+where
+    Value: From<T>,
+{
     fn from(guard: ThreadGuard<T>) -> Self {
-        Self::from(guard.get_ref())
+        Self::from(guard.into_inner())
     }
 }
 ```